### PR TITLE
feat(retros-in-disguise): added standup activity details

### LINKED ISSUE #7970: RiD Details View: Check-in Template Details (no sidebar)
Implement the template portion of RiD Details View for Check-ins (i.e. no sidebar).

[Figma](https://www.figma.com/file/PmE9A2ryODVzFPjZFJn7qi/Parabol-Activity-Library?node-id=310%3A23597&t=fROVwP1DKIkIBZJX-1)

AC:

- [ ] Clicking the check-in activity in the activity library navigates to the details view for check-ins
- [ ] View matches designs
- [ ] URL is unique to the activity and can be copied+shared+used on landing pages

### PR DESCRIPTION
**Pull request by BartoszJarocki**
# Description

Fixes #7970 #7969 

This PR also splits `ActivityDetails` component into smaller chunks. It's not meant to be final refactoring, but some parts had to be changed to support showing not only templates but meeting details.

## Demo

https://github.com/ParabolInc/parabol/assets/1017620/25450b5e-d551-4279-823c-2f76bdd2a476

## Testing scenarios

[Please list all the testing scenarios a reviewer has to check before approving the PR]

- [ ] Open activity library, navigate to standup details
- [ ] Open activity library, navigate to check-in details
- [ ] Smoke test showing retros and planning details to make sure refactoring didn't break anything

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx
Comment (BartoszJarocki) [inline packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx:44]: This pretty much untouched and could be a subject to further refactoring. For now, it was just taken out from the main component, to allow displaing other UI needed for meetings.

File: packages/client/components/ActivityLibrary/ActivityDetails/MeetingDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/MeetingDetails.tsx:1]: +1 This is effectively the same as `TemplateDetails` with some tweaks, right? Why not merge the two components and do conditional renderings if the activity isn't a template? That would ensure consistency between generic activity details for template activities and generic activity details for non-template activities. I.e. if we changed anything about how generic details are rendered, we'd need to change that both here and in `TemplateDetails`, instead of in just one place.

File: packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx:135]: what component are you specifically talking about? two things are extracted, one component for meeting details, one for template details

**Comment by BartoszJarocki** [comment]
> I think this refactor is trying to abstract too much. Specifically, I think the way this is creating a hook to try to abstract a lot of logic + component renderings is the wrong way to go about this refactor, and I think `MeetingDetails` and `TemplateDetails` are similar enough to be a single component that conditionally renders certain elements based on whether it's a template activity.
> 
> I'd suggest merging a lot of these separated files back into a single component that depends on conditional rendering, and instead extract smaller chunks of the activity details (e.g. extract an [`IntegrationTips` component](https://github.com/ParabolInc/parabol/blob/8686edd8ca1c9f1bbc84846be69b33457cebe42f/packages/client/components/ActivityLibrary/ActivityDetails.tsx#L363-L374), separate components for [retro prompt + poker dimension lists](https://github.com/ParabolInc/parabol/blob/8686edd8ca1c9f1bbc84846be69b33457cebe42f/packages/client/components/ActivityLibrary/ActivityDetails.tsx#L377-L404), and a [`TemplateActions` component](https://github.com/ParabolInc/parabol/blob/8686edd8ca1c9f1bbc84846be69b33457cebe42f/packages/client/components/ActivityLibrary/ActivityDetails.tsx#L285-L345))

I'll think about extracting smaller parts, but merging it all into one big file is not an option in my opinion. logic related to templates is much more complex and unrelated to meetings without templates (editing, handling ownership, prompts, data dependencies). the way it was done, makes it _really_ hard for someone new understand what's happening there. This PR is already big, and should not aim to fix all the problems with current implementation. Will think what can I do to improve this.

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:80]: what I think is wrong, is mixing mapping the data and including UI components alongside of it. Mapping from meeting/template to `ActivityDetails` is something that fixes not unified data model and I think is correct. Any thoughts about removing react components from this, and rendering the react components outside of this hook, based on data returned here? would that be ok?

File: packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx
Comment (jmtaber129) [reply packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx:135]: Sorry, I should have been more specific. I'm referring to the poker dimensions + retro prompts [here](https://github.com/ParabolInc/parabol/pull/8199/files#diff-66a34f730f42433e7b2b47ef1b1dbe15d1526a70da8f31e31aab44b532751614R137-R163). This template scale portal logic is pretty coupled with template dimensions, so we should try to keep this colocated with that.

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:80]: thank you!

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:63]: seems like a good idea, will do and see if there are no issues with types

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [inline packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:79]: by doing this, we know if it's a template, then there's a template object defined and we don't have to use `!` or `?` when using it

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:134]: +1 Let's get rid of this, since this is really just a convenience for `!!activity.template`

File: packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx:260]: +1 This should really be using the `activity.description` returned by `useActivityDetails`, right?

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:106]: +1 I don't think this is actually used anywhere as-is, as anywhere the description is rendered, we're doing `ACTIVITY_TYPE_DATA_LOOKUP.description[type]` instead. We should either a) use this field in the rendering (my preference), or b) remove this field from the activity config.

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:20]: +1 Let's flip the mappings here, so it's `MeetingTypeEnum -> field -> data` instead of `field -> MeetingTypeEnum -> data`

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (jmtaber129) [inline packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:88]: +1 No need for `edit` here

**Review by jmtaber129 — APPROVED** [review]
+1 The APIs here are still a little messy, since some data comes from the activity returned by `useActivityDetails`, while other data comes from lookups for `type`. Let's change the `MeetingDetails` and `TemplateDetails` components to take the full `activity` returned by `useActivityDetails`. Then, the API would be "render the details based on the data provided for this activity" vs "render details given these specific pieces of data, and looking up details as needed".

Not merge-blocking though. Nice job on the refactor!

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:134]: Just like I said [here](https://github.com/ParabolInc/parabol/pull/8199#discussion_r1198874955), TS still complains about using `activity.template` in ternaries after doing `!!activity.template` so I'd rather keep the code more readable instead of using `?` and `!` in places where it's not actually needed.

**Comment by BartoszJarocki** [comment]
@jmtaber129 pretty much all fixed, good suggestions! moving to maintainer review

**Review by mattkrick — APPROVED** [review]
+1 looks good! just a few type changes & var names

File: packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx:28]: yes, good catch!

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:95]: I'd say it should be more like `noTemplatesActivityIds`

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:64]: yes that's cleaner, thank you!

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:77]: Screenshot 2023-05-23 at 14 12 10

we can remove it, but it was meant to provide information visible above (from the designs)

File: packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx
Comment (BartoszJarocki) [reply packages/client/components/ActivityLibrary/ActivityDetails/TemplateDetails.tsx:137]: I resolved conflicts but this seems to be still needed for `useTemplateDescription`.  I've added `ActivityDetailsBadges` usage here and left `MeetingDetails` untouched as I suspect we can't change the category for that and it'll be hardcoded for a while. Do you mind checking if everything you did works as expected. I did and I think it's all good, just want to make sure. Radix menus are 🚀 ❤️  btw!

File: packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx
Comment (mattkrick) [reply packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx:28]: ohhhh! now I get it.
Could we instead give `useActivityDetails` its own fragment? that way we won't have to know to remove it if the hook gets refactored.

File: packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx
Comment (mattkrick) [reply packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx:133]: TIL that discriminating on the presence of an object isn't supported... weird! i'll bump ts & see if v5 supports it, I could've sworn it did...

File: packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx
Comment (mattkrick) [reply packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx:28]: i'll write a patch & see if you like it

Comment (mattkrick) [reply packages/client/components/ActivityLibrary/ActivityDetails/ActivityDetails.tsx:28]: i noticed there are a few other fields & fragments that are overfetched here, I don't want this to block the release so we'll merge it anyways, but we want to go back & fix this to make the code a little easier to understand while it's still fresh in our heads so it's easier for the folks in the future to work on

**Pull request merged**
